src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -96,30 +96,39 @@ function addModeListener() {
 }
 
 /*******************************************************************************
- * TOC interactivity
+ * Right sidebar table of contents (TOC) interactivity
  */
+function setupPageTableOfContents() {
+  const pageToc = document.querySelector("#pst-page-toc-nav");
+  pageToc.addEventListener("click", (event) => {
+    if (!event.target.matches(".nav-link")) {
+      return;
+    }
+    const clickedLink = event.target;
 
-/**
- * TOC sidebar - add "active" class to parent list
- *
- * Bootstrap's scrollspy adds the active class to the <a> link,
- * but for the automatic collapsing we need this on the parent list item.
- *
- * The event is triggered on "window" (and not the nav item as documented),
- * see https://github.com/twbs/bootstrap/issues/20086
- */
-function addTOCInteractivity() {
-  window.addEventListener("activate.bs.scrollspy", function () {
-    const navLinks = document.querySelectorAll(".bd-toc-nav a");
-
-    navLinks.forEach((navLink) => {
-      navLink.parentElement.classList.remove("active");
+    // First, clear all the added classes and attributes
+    // -----
+    pageToc.querySelectorAll("a[aria-current]").forEach((el) => {
+      el.removeAttribute("aria-current");
     });
-
-    const activeNavLinks = document.querySelectorAll(".bd-toc-nav a.active");
-    activeNavLinks.forEach((navLink) => {
-      navLink.parentElement.classList.add("active");
+    pageToc.querySelectorAll(".active").forEach((el) => {
+      el.classList.remove("active");
     });
+
+    // Then add the classes and attributes to where they should go now
+    // -----
+    clickedLink.setAttribute("aria-current", "true");
+    clickedLink.classList.add("active");
```

Comment:
This will add styles to the link that make it look like the current link. This matches the behavior of the GitHub Actions docs, for example see [Understanding GitHub Actions](https://docs.github.com/en/actions/about-github-actions/understanding-github-actions#next-steps). 

Note that once the user clicks on the link, it keeps this styling even if they scroll to a completely different section. It only gets cleared and applied to a different TOC link if the user clicks on a different TOC link.